pennylane/spin/lattice.py

```diff
@@ -190,7 +206,111 @@ def _generate_grid(self, neighbour_order):
             lattice_points.append(point)
             lattice_map.append(node_index)
 
-        return math.array(lattice_points), math.array(lattice_map)
+        return math.array(lattice_points), lattice_map
+
+    def _get_custom_edges(self, custom_edges, lattice_map):
+        """Generates the edges described in `custom_edges` for all unit cells.
+
+        Args:
+          custom_edges (Optional[list(list(tuples))]): Specifies the edges to be added in the lattice.
+              Default value is None, which adds the edges based on neighbour_order.
+              Each element in the list is for a separate edge, and can contain 1 or 2 tuples.
+              First tuple contains the index of the starting and ending vertex of the edge.
+              Second tuple is optional and contains the operator on that edge and coefficient
+              of that operator.
+          lattice_map (list[int]): A list to represent the node number for each lattice_point.
+
+        Returns:
+          List of edges.
+
+        **Example**
+
+        Generates a square lattice with a single diagonal and assigns a different operation
```

Comment:
This example seems to be for the Lattice class not for this function.

Reply:
This is not a standalone function, it works with lattice class, as in to define the edges for the lattice class in a custom way. Lattice class generates edges based on the distance between the vertices, whereas if we pass custom_edges then the edges are generated based on what is provided.

Reply:
I prefer to see the exact input and output of this function. The current example is irrelevant.